lib/_tls_common.js

```diff
@@ -153,6 +153,24 @@ exports.createSecureContext = function createSecureContext(options) {
     }
   }
 
+  const { engineKey, privateKeyEngine } = options;
+  if (engineKey !== undefined && privateKeyEngine !== undefined) {
+    if (typeof engineKey === 'string' && typeof privateKeyEngine === 'string') {
+      if (c.context.setEngineKey)
```

Comment:
As per my comment below, I don't see how you get a false here unless you're manually messing with the binding as in the test.

Does `OPENSSL_NO_ENGINE` help us at all? Maybe you can wrap the `"setEngineKey"` in an `ifndef` for that like `setClientCertEngine` is?

Reply:
> Maybe you can wrap the "setEngineKey" in an ifndef for that like setClientCertEngine is?

Yes, probably I forgot this, thank you, will fix. Can't tell what `OPENSSL_NO_ENGINE` is in general for though.

Reply:
It's describes when OpenSSL is built without engine capability.

Reply:
Yes, `setEngineKey` is added conditionally now:

https://github.com/nodejs/node/pull/28973/files#diff-801e3948990f4965a8ea4aca4a423864R475